pydra/engine/specs.py

```diff
@@ -360,9 +359,23 @@ def _field_metadata(self, fld, inputs, output_dir):
         if "value" in fld.metadata:
             return output_dir / fld.metadata["value"]
         elif "output_file_template" in fld.metadata:
-            return output_dir / fld.metadata["output_file_template"].format(
-                **inputs.__dict__
+            sfx_tmpl = (output_dir / fld.metadata["output_file_template"]).suffixes
```

Comment:
how do you detect suffixes. for imaging files or other modalities this may need to be quite custom? for example `nii.gz`, `BRIK/HEAD`, etc.,.

how will pydra generalize this across domains?

Reply:
my main problem was that `input_1` could be `filename`, `filename.nii` or `filename.nii.gz` and output should be always `filename.nii.gz`, and the templaete is `[input_1].nii.gz`...
so I decided to remove all suffixes if template has it's own, happy to hear better suggestions

Reply:
is that specific to bosh or boutiques that output is always compressed nifti? also does bosh/boutiques enforce nifti as output? i don't think anything in boutiques enforces it.

Reply:
if you see example for [bet](https://zenodo.org/record/1482743/files/fsl_bet.json?download=1) - you will see that "output-files" has "path-template": "[MASK].nii.gz", but `maskfile` (that has "value-key": "[MASK]"), could be either `filename` or `filename.nii.gz`.  For both cases  `output-files` is `filename.nii.gz`

Reply:
that may be true of bet, but boutiques can support other tools as well. so there needs to be a general way of deciding what the outputs would be.

this is the part where knowing what outputs should be created given the inputs plays a role, but i don't think that's captured in boutiques yet. so as a first pass, you can simply leave inputs/outputs as separate things.

Reply:
I understood that it's not always a case, but I thought this is a good start. Not sure how do you want me to keep it separately. From the zenodo file I have only a template that uses inputs values